pkg/plugins/policies/core/matchers/dataplane.go

```diff
@@ -70,37 +74,113 @@ func MatchedPolicies(rType core_model.ResourceType, dpp *core_mesh.DataplaneReso
+func inboundsSelectedByPolicy(
+	policyResource core_model.Resource,
+	dpp *core_mesh.DataplaneResource,
+	gateway *core_mesh.MeshGatewayResource,
+) []core_rules.InboundListener {
+	policy := policyResource.GetSpec().(core_model.Policy)
+
+	switch policy.GetTargetRef().Kind {
 	case common_api.Mesh:
 		return inboundsSelectedByTags(nil, dpp, gateway)
 	case common_api.MeshSubset:
-		return inboundsSelectedByTags(tr.Tags, dpp, gateway)
+		return inboundsSelectedByTags(policy.GetTargetRef().Tags, dpp, gateway)
 	case common_api.MeshService:
 		return inboundsSelectedByTags(map[string]string{
-			mesh_proto.ServiceTag: tr.Name,
+			mesh_proto.ServiceTag: policy.GetTargetRef().Name,
 		}, dpp, gateway)
 	case common_api.MeshServiceSubset:
 		tags := map[string]string{
-			mesh_proto.ServiceTag: tr.Name,
+			mesh_proto.ServiceTag: policy.GetTargetRef().Name,
 		}
-		for k, v := range tr.Tags {
+		for k, v := range policy.GetTargetRef().Tags {
 			tags[k] = v
 		}
 		return inboundsSelectedByTags(tags, dpp, gateway)
+	case common_api.MeshHTTPRoute:
+		rr, ok := policyResource.(*core_rules.ResolvedResource)
+		if !ok {
+			panic(fmt.Sprintf("provided policy is referencing %v but not resolved", common_api.MeshHTTPRoute))
+		}
+		mhr, ok := rr.ResolvedTargetRefs[policy.GetTargetRef().Hash()]
+		if !ok {
+			panic(fmt.Sprintf("can't resolve %v targetRef", common_api.MeshHTTPRoute))
+		}
+		return inboundsSelectedByPolicy(mhr, dpp, gateway)
 	default:
 		return []core_rules.InboundListener{}
 	}
 }
 
+func resolveTargetRefs(rl []core_model.Resource, resources xds_context.Resources) ([]core_model.Resource, error) {
+	rv := []core_model.Resource{}
+
+	for _, r := range rl {
+		policy := r.GetSpec().(core_model.Policy)
+
+		switch policy.GetTargetRef().Kind {
+		case common_api.MeshHTTPRoute:
+			mhr, err := resolveMeshHTTPRouteRef(r.GetMeta(), policy.GetTargetRef().Name, resources)
+			if err != nil {
+				return nil, err
+			}
+			rv = append(rv, &core_rules.ResolvedResource{
+				Resource: r,
+				ResolvedTargetRefs: map[common_api.TargetRefHash]core_model.Resource{
+					policy.GetTargetRef().Hash(): mhr,
+				},
+			})
+		default:
+			rv = append(rv, r)
+		}
+	}
+
+	return rv, nil
+}
+
+func resolveMeshHTTPRouteRef(refMeta core_model.ResourceMeta, refName string, resources xds_context.Resources) (*meshhttproute_api.MeshHTTPRouteResource, error) {
+	mhrs := resources.ListOrEmpty(meshhttproute_api.MeshHTTPRouteType)
+	for _, item := range mhrs.GetItems() {
+		if isReferenced(refMeta, refName, item.GetMeta()) {
+			return item.(*meshhttproute_api.MeshHTTPRouteResource), nil
+		}
+	}
+	return nil, errors.Errorf("MeshHTTPRoute %v not found", refName)
+}
+
+func isReferenced(refMeta core_model.ResourceMeta, refName string, resourceMeta core_model.ResourceMeta) bool {
+	if len(refMeta.GetNameExtensions()) == 0 {
+		return refName == resourceMeta.GetName()
```

Comment:
I'd rather expect to compare only name if `resourceMeta` doesn't have a `Namespace`, is this correct if `resourceMeta` has a `Namespace` but `refMeta` doesn't?